### PR TITLE
Translate VSTS Git urls to GitHub URLs

### PR DESCRIPTION
**Pull request by pranavkm**
Possible fix for https://github.com/aspnet/AspNetCore/issues/7710

### CODE REVIEW

File: Directory.Build.targets [hunk 1]
```diff
@@ -98,6 +98,35 @@
...
+    <!-- The convention for names of Azure DevOps repositories mirrored from GitHub is "{GitHub org name}-{GitHub repository name}" -->
+    <PropertyGroup>
+      <!-- There are quite a few git repo forms:
+        https://dnceng@dev.azure.com/dnceng/internal/_git/dotnet-arcade-services
+        https://dev.azure.com/dnceng/internal/_git/dotnet-arcade-services
+        https://dnceng.visualstudio.com/internal/_git/dotnet-arcade-services
+        dnceng@vs-ssh.visualstudio.com:v3/dnceng/internal/dotnet-arcade-services
+        git@ssh.dev.azure.com:v3/dnceng/internal/dotnet-arcade-services
+      -->
+      <_Pattern>(https://dnceng%40dev\.azure\.com/dnceng/internal/_git|https://dev\.azure\.com/dnceng/internal/_git|https://dnceng\.visualstudio\.com/internal/_git|dnceng%40vs-ssh\.visualstudio\.com:v3/dnceng/internal|git%40ssh\.dev\.azure\.com:v3/dnceng/internal)/([^/-]+)-(.+)</_Pattern>
```
Comment (Eilon) [inline Directory.Build.targets]: Lovely ☹️

**Review by Eilon — COMMENTED** [review]
The regex looks good to me, assuming it works in practice. But, I'm wondering, what do other repos do? Does everyone need to do this?

File: Directory.Build.targets [hunk 2]
```diff
@@ -98,6 +98,35 @@
 
   </ItemGroup>
 
+  <!-- Copied from https://github.com/dotnet/arcade/blob/9d0fd805448082c8d55e2434607b481bca70a146/src/Microsoft.DotNet.Arcade.Sdk/tools/RepositoryInfo.targets#L12-L38 -->
+  <Target Name="_TranslateAzureDevOpsUrlToGitHubUrl"
+          DependsOnTargets="$(SourceControlManagerUrlTranslationTargets)"
+          BeforeTargets="SourceControlManagerPublishTranslatedUrls">
```
Comment (natemcmaster) [inline Directory.Build.targets]: Where is SourceControlManagerPublishTranslatedUrls? Part of sourcelink?

**Review by natemcmaster — APPROVED** [review]
If it works, :shipit:

**Comment by natemcmaster** [comment]
> Does everyone need to do this?

Everyone else is using the Arcade SDK 😏, so probably not

File: Directory.Build.targets (hunk 2 above)
Comment (pranavkm) [reply Directory.Build.targets]: Yup - https://github.com/dotnet/sourcelink/blob/23818e7e498e5e9b3b7f585fb8ac887c3077706d/src/SourceLink.Common/build/InitializeSourceControlInformation.targets#L28

**Comment by Eilon** [comment]
Is the real fix to spend effort on moving to Arcade? If so, then I'd rather spend cycles on that.

**Comment by Eilon** [comment]
And we need Arcade for source-build anyway.

**Comment by pranavkm** [comment]
@aspnet-hello triage https://dev.azure.com/dnceng/public/_build/results?buildId=153557&view=logs

**Comment by aspnet-hello** [comment]
This comment was made automatically. If there is a problem contact ryanbrandenburg.

I've triaged the above build. I've created/commented on the following issue(s)
https://github.com/aspnet/AspNetCore-Internal/issues/2250

**Pull request merged**